src/app/conditional-rendering/page.tsx

```diff
@@ -10,21 +10,28 @@ function Component3() {
   return <>コンポーネント3</>;
 }
 
+const CONDITIONS = {
+  ADMIN: 'Admin',
+  USER: 'User',
+  GUEST: 'Guest',
+};
+
+type Condition = typeof CONDITIONS[keyof typeof CONDITIONS];
+
 type Props = {
-  condition: string;
+  condition: Condition;
 };
 
 /**
  * 条件分岐が多い場合はデータの持ち方を工夫しましょう。
  * @returns
  */
 export default function Page({ condition }: Props) {
-  switch (condition) {
-    case 'Admin':
-      return <Component1 />;
-    case 'User':
-      return <Component2 />;
-    case 'Guest':
-      return <Component3 />;
-  }
+  const conditionPairs = {
+    [CONDITIONS.ADMIN]: <Component1 />,
+    [CONDITIONS.USER]: <Component2 />,
+    [CONDITIONS.GUEST]: <Component3 />,
+  };
+
+  return conditionPairs[condition];
```

Comment:
propsの指定などもできるので、👇のように書いた方がオススメ。

```suggestion
  const conditionPairs = {
    [CONDITIONS.ADMIN]: <Component1 />,
    [CONDITIONS.USER]: <Component2 />,
    [CONDITIONS.GUEST]: <Component3 />,
  };

  const Component = conditionPairs[condition]
  return <Component />;
```

Reply:
ありがとうございます！
[c963bd4](https://github.com/koguuum/react-bad-codes/pull/1/commits/c963bd4ac2a2b093902231d13e26e3a348575cff)で修正しました！